back/src/domain/immersionOffer/useCases/discussions/AddExchangeToDiscussionAndTransferEmail.ts

```diff
@@ -0,0 +1,193 @@
+import { z } from "zod";
+import { DiscussionId, ExchangeRole } from "shared";
+import {
+  BadRequestError,
+  NotFoundError,
+} from "../../../../adapters/primary/helpers/httpErrors";
+import {
+  UnitOfWork,
+  UnitOfWorkPerformer,
+} from "../../../core/ports/UnitOfWork";
+import { TransactionalUseCase } from "../../../core/UseCase";
+import { SaveNotificationAndRelatedEvent } from "../../../generic/notifications/entities/Notification";
+import {
+  addExchangeToDiscussion,
+  createOpaqueEmail,
+  ExchangeEntity,
+} from "../../entities/DiscussionAggregate";
+
+type BrevoAttachment = {
+  Name: string;
+  ContentType: string;
+  ContentLength: number;
+  ContentID: string;
+  DownloadToken: string;
+};
+
+type BrevoRecipient = {
+  Name: string | null;
+  Address: string;
+};
+
+type BrevoEmailItem = {
+  Uuid: string[];
+  MessageId: string;
+  InReplyTo: string | null;
+  From: BrevoRecipient;
+  To: BrevoRecipient[];
+  Cc: BrevoRecipient[];
+  ReplyTo: BrevoRecipient | null;
+  SentAtDate: string;
+  Subject: string;
+  Attachments: BrevoAttachment[];
+  RawHtmlBody: string | null;
+  RawTextBody: string | null;
+};
+
+export type BrevoInboundBody = {
+  items: BrevoEmailItem[];
+};
+
+const brevoInboundBodySchema: z.Schema<BrevoInboundBody> = z.object({
+  items: z.array(
+    z.object({
+      Uuid: z.array(z.string()),
+      MessageId: z.string(),
+      InReplyTo: z.string().nullable(),
+      From: z.object({
+        Name: z.string().nullable(),
+        Address: z.string(),
+      }),
+      To: z.array(
+        z.object({
+          Name: z.string().nullable(),
+          Address: z.string(),
+        }),
+      ),
+      Cc: z.array(
+        z.object({
+          Name: z.string().nullable(),
+          Address: z.string(),
+        }),
+      ),
+      ReplyTo: z
+        .object({
+          Name: z.string().nullable(),
+          Address: z.string(),
+        })
+        .nullable(),
+      SentAtDate: z.string(),
+      Subject: z.string(),
+      Attachments: z.array(
+        z.object({
+          Name: z.string(),
+          ContentType: z.string(),
+          ContentLength: z.number(),
+          ContentID: z.string(),
+          DownloadToken: z.string(),
+        }),
+      ),
+      RawHtmlBody: z.string().nullable(),
+      RawTextBody: z.string().nullable(),
+    }),
+  ),
+});
+
+export class AddExchangeToDiscussionAndTransferEmail extends TransactionalUseCase<BrevoInboundBody> {
+  inputSchema = brevoInboundBodySchema;
```

Comment:
```suggestion
  inputSchema = brevoInboundBodySchema;
```
Sous le constructeur